### ISSUE TITLE
Using getProvisioningProfile fails with conflict: You can't enable both :development and :adhoc

### ISSUE THREAD

**Issue by dachev** [fastlane/fastlane#12506]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

`getProvisioningProfile(appIdentifier: app_identifier, username: appleID)`

fails with

> You can't enable both :development and :adhoc

If I delete these arguments from the Swift source, the command fails with

> You can't enable both :force and :readonly

When this pair is removed too, the function call completes successfully. This has been reported and ignored multiple times (GH-11519, GH-12053, GH-6642) It looks like an argument validation bug.

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>
[✔] 🚀 
[00:44:59]: fastlane detected a Gemfile in the current directory
[00:44:59]: however it seems like you don't use `bundle exec`
[00:44:59]: to launch fastlane faster, please use
[00:44:59]: 
[00:44:59]: $ bundle exec fastlane ios beta app_identifiers:app_identifier
[00:44:59]: 
[00:44:59]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[00:45:00]: loading manifest: /Users/user/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/fastlane-2.95.0/fastlane/swift/upgrade_manifest.json
[00:45:00]: FastlaneSwiftRunner project is up-to-date
[00:45:00]: $ xcodebuild -showBuildSettings -scheme FastlaneRunner -project ./fastlane/swift/FastlaneSwiftRunner/FastlaneSwiftRunner.xcodeproj
[00:45:02]: $ set -o pipefail && xcodebuild -scheme FastlaneRunner -project ./fastlane/swift/FastlaneSwiftRunner/FastlaneSwiftRunner.xcodeproj -destination 'generic/platform=macOS' | tee /Users/user/Library/Logs/gym/App-FastlaneRunner.log | xcpretty
[00:45:05]: $ ./fastlane/FastlaneRunner lane beta app_identifiers app_identifier > /dev/null
[00:45:05]: ------------------------------------
[00:45:05]: --- Step: increment_build_number ---
[00:45:05]: ------------------------------------
Current version of project ProjectName is: 
    19

/path/to/project
[00:45:06]: $ cd /path/to/project && agvtool next-version -all && cd -
[00:45:07]: ▸ Setting version of project ProjectName to:
[00:45:07]: ▸ 20.
[00:45:07]: ▸ Also setting CFBundleVersion key (assuming it exists)
[00:45:07]: ▸ Updating CFBundleVersion in Info.plist(s)...
[00:45:07]: ▸ Updated CFBundleVersion in "ProjectName.xcodeproj/../ProjectName/Info.plist" to 20
[00:45:07]: ▸ /path/to/project
[00:45:08]: -----------------------------------
[00:45:08]: --- Step: update_app_identifier ---
[00:45:08]: -----------------------------------
[00:45:08]: Updated ProjectName.xcodeproj 💾.
[00:45:08]: ------------------------------
[00:45:08]: --- Step: get_certificates ---
[00:45:08]: ------------------------------

+---------------+------------------------------------+
|              Summary for cert 2.95.0               |
+---------------+------------------------------------+
| development   | false                              |
| force         | false                              |
| username      | user@domain.com                   |
| team_id       | 3EJK83E37K                         |
| keychain_path | ~/Library/Keychains/login.keychain |
| platform      | ios                                |
+---------------+------------------------------------+

[00:45:08]: Starting login with user 'user@domain.com'
[00:45:10]: Successfully logged in
[00:45:12]: Found the certificate QY5E9GUFKW (iOS Distribution) which is installed on the local machine. Using this one.
[00:45:13]: Verifying the certificate is properly installed locally...
[00:45:13]: Successfully installed certificate QY5E9GUFKW
[00:45:13]: Use signing certificate 'QY5E9GUFKW' from now on!
[00:45:13]: --------------------------------------
[00:45:13]: --- Step: get_provisioning_profile ---
[00:45:13]: --------------------------------------
[00:45:13]: Error resolving conflict between options: 'adhoc' and 'development'
[00:45:13]: You passed invalid parameters to 'get_provisioning_profile'.
[00:45:13]: Check out the error below and available options by running `fastlane action get_provisioning_profile`
+---------------------+-----------------------------------------------------------------------------------+
|                                              Lane Context                                               |
+---------------------+-----------------------------------------------------------------------------------+
| BUILD_NUMBER        | 20                                                                                |
| CERT_FILE_PATH      | /path/to/project/QY5E9GUFKW.  |
|                     | cer                                                                               |
| CERT_CERTIFICATE_ID | QY5E9GUFKW                                                                        |
+---------------------+-----------------------------------------------------------------------------------+
[00:45:13]: You can't enable both :development and :adhoc

+------+--------------------------+-------------+
|               fastlane summary                |
+------+--------------------------+-------------+
| Step | Action                   | Time (in s) |
+------+--------------------------+-------------+
| 1    | increment_build_number   | 2           |
| 2    | update_app_identifier    | 0           |
| 3    | get_certificates         | 4           |
| 💥   | get_provisioning_profile | 0           |
+------+--------------------------+-------------+

[00:45:13]: fastlane finished with errors

[!] You can't enable both :development and :adhoc
  </pre>
</details>

### Environment

<details>
  <pre>
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.4                                     |
| Ruby                        | 2.3.0                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.10.2                          |
| Installation Source         | ~/.rbenv/versions/2.3.0/bin/fastlane        |
| Host                        | Mac OS X 10.13.4 (17E199)                   |
| Ruby Lib Dir                | ~/.rbenv/versions/2.3.0/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2h  3 May 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.3                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile.swift`</summary>

```ruby
// This file contains the fastlane.tools configuration
// You can find the documentation at https://docs.fastlane.tools
//
// For a list of all available actions, check out
//
//     https://docs.fastlane.tools/actions
//

import Foundation

class Fastfile: LaneFile {
    let default_app_identifiers = ["default_app_identifier_1", "default_app_identifier_2"]
  
    func betaLane(withOptions options: [String: String]?) {
        desc("Create and deploy beta builds to TestFlight")
        
        incrementBuildNumber(buildNumber: options?["build_number"])
        
        var app_identifiers = default_app_identifiers
        if let arg = options?["app_identifiers"] {
            app_identifiers = arg.components(separatedBy: ",")
        }
        
        for app_identifier in app_identifiers {
            updateAppIdentifier(xcodeproj: "ProjectName.xcodeproj", plistPath: "ProjectName/Info.plist", appIdentifier: app_identifier)
            getCertificates(username: appleID, teamId: teamID, keychainPath: "~/Library/Keychains/login.keychain")
            getProvisioningProfile(appIdentifier: app_identifier, username: appleID)
            buildApp(workspace: "ProjectName.xcworkspace", scheme: "ProjectName")
            uploadToTestflight(username: appleID, appIdentifier: app_identifier, teamId: itcTeam)
        }
    }
}

```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.95.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| slack-notifier            | 2.3.2        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| addressable               | 2.5.2        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| public_suffix             | 2.0.5        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.13.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| bundler                   | 1.16.1       |
| faraday_middleware        | 0.12.2       |
| naturally                 | 2.1.0        |
| simctl                    | 1.6.4        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| emoji_regex               | 0.1.1        |
| io-console                | 0.4.6        |
| excon                     | 0.62.0       |
| CFPropertyList            | 3.0.0        |
| mini_portile2             | 2.3.0        |
| nokogiri                  | 1.8.1        |
| plist                     | 3.4.0        |
| atomos                    | 0.1.2        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.5        |
| xcodeproj                 | 1.5.8        |
| faraday                   | 0.15.1       |
| unf_ext                   | 0.0.7.5      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20180417 |
| fastimage                 | 2.1.3        |
| json                      | 2.1.0        |
</details>

*generated on:* **2018-05-12**
</details>
  </pre>
</details>

**Comment by txuslee** [comment]
If it can be of any help, when executed the task through command line like:

`fastlane run get_provisioning_profile app_identifier:bundleID username:appleID`

It works for the same username and app_identifier where it fails in swift.

**Comment by jakubholik** [comment]
Having exactly same problem running the action from within the Fastfile

**Comment by txuslee** [comment]
Still reproducible on:

fastlane 2.98.0 & FastlaneRunnerAPIVersion [0.9.21]

**Comment by dachev** [comment]
Please don't close this issue again. It's one thing not make progress because you have your hands full, but closing an issue 4 times is just insulting.

**Comment by dachev** [comment]
@doruvil I ran into other problems and ended up switching to Ruby. Thanks for the heads up though.

**Comment by doruvil** [comment]
@joshdholtz maybe is there a way to integrate this change in one of the following version of fastlane? Or this is blocked by something else?
